src/ui/zcl_abapgit_gui_view_repo.clas.abap

```diff
@@ -536,6 +551,13 @@ CLASS zcl_abapgit_gui_view_repo IMPLEMENTATION.
       WHEN c_actions-display_more.      " Increase MAX lines limit
         mv_max_lines    = mv_max_lines + mv_max_setting.
         ev_state        = zif_abapgit_definitions=>c_event_state-re_render.
+      WHEN c_actions-zip_import2_load.
+        zcl_abapgit_zip=>import( iv_key = mo_repo->get_key( ) iv_no_deserialize = abap_true ).
+        ev_state        = zif_abapgit_definitions=>c_event_state-re_render.
+      WHEN c_actions-zip_import2_apply.
+        zcl_abapgit_services_repo=>gui_deserialize( mo_repo ).
+        mo_repo->reset_remote( ).
```

Comment:
suggest not resetting after apply, in order to show the user the diff if something did not load successfully, this is the behavior for online repos

Reply:
Interesting point. But then it is better to check if the deserialize was successful. Is the a way ?

Reply:
suggest just showing the user the same behavior as for online repos, after import it shows a diff, and it is possible to trigger pull again

Reply:
to this and the `reset_remote` comment below.
Actually started to adjust the code but realized that the UI logic gets worse. Here is some arguments:

1) there must be an action to reset the imported zip (to cancel the set_remote). This is "Refresh" button. Most intuitive. Refresh button calls repo->refresh...
2) ... online repo does overwrite remotes on refresh ...
3) ... so reset_remote really fits well in refresh, it does not violate current online logic
4) without `mo_repo->reset_remote( )` on apply the UI hangs in "Apply" state and with .abapgit.xml displayed, without diffs, just "white" lines. This looks a bit strange.
5) Option to "restart" on wrong serialization is a good point. But it is very easy to reload zip to see the "unfinished" objects. And apart from online repo offline does not store ref to commit number so reloading zip will just show new real picture. 

So I'd say let's leave both `reset_remote`. Without it UI really feel less logical. I can remove them and commit for a try. But imho it feels worse.

Reply:
1. why is this needed? The user can import a new zip at any time?
5. but it will be 2 steps, import zip => pull changes => import zip => show diff. Whereas online is pull changes => show diff

Reply:
1) To cancel back the action. The import button disappears in fact. So the flow is not ambiguous. You start the import - finish it or cancel it. I can leave the import button but that is less clear and more cluttered. IMHO. Here is now it looks now

![2018-09-26_16h37_18](https://user-images.githubusercontent.com/15635498/46084466-65210a80-c1ac-11e8-9594-635e90d593f4.gif)


5) Hmm, I partially agree. But how often does it happen that import fails ? Good "Automation" for all the existing failures is impossible. 2 scenarios: "Normal" - everything is OK, ag showed activation dialog blah blah, and repo looks line normal offline repo, "as usual". "Failure" - ag (actually sap) will show some error messages by the way which will be noticed by user and he can take actions.

OK. I have another idea for 5. If we import zip and there are no changes then a warning is displayed "no difference with zip detected" (maybe not warning, maybe it's green ..  but I mean the box above the repo). So ... when I apply zip and everything is successful then this message also shown (remote is not reset). Also if there are no diffs then "import zip" is displayed and apply+show_diff are not. However this might be a bit tricky to implement. And also not quite symetric with online. Online does not show any messages. So will be more conditioning inside. E.g. menu construction will have to know if there are diffs in a repo ... hmmmm

Reply:
One more problem: offline does not have checksums by default. This makes all the files MM unless manual update checksums executed.

Reply:
checksums: should also align this so it works the same for online and offline, but we can do this later in a follow up PR